library/ziti_ctrl.c

```diff
@@ -84,8 +84,14 @@ static void ctrl_default_cb (void *s, ziti_error *e, struct ctrl_resp *resp) {
 }
 
 static void ctrl_login_cb(ziti_session *s, ziti_error *e, struct ctrl_resp *resp) {
-    resp->ctrl->session = strdup(s->token);
-    um_http_header(&resp->ctrl->client, "zt-session", s->token);
+    if (e) {
+        ZITI_LOG(ERROR, "%s(%s)", e->code, e->message);
+    }
+
+    if (s) {
+        resp->ctrl->session = strdup(s->token);
+        um_http_header(&resp->ctrl->client, "zt-session", s->token);
+    }
     ctrl_default_cb(s, e, resp);
 }
```

Comment:
should this be calling free_ziti_err ?

Reply:
it is freed by the client code (in client callback)